runtime/parachains/src/router/hrmp.rs

```diff
@@ -0,0 +1,1244 @@
+				.map(|(k, _)| k.sender)
+				.collect::<HashSet<_>>(),
+		);
+		for (open_channel_initiator, expected_num) in
+			<Router as Store>::HrmpOpenChannelRequestCount::iter()
+		{
+			let actual_num = <Router as Store>::HrmpOpenChannelRequests::iter()
+				.filter(|(ch, _)| ch.sender == open_channel_initiator)
+				.count() as u32;
+			assert_eq!(expected_num, actual_num);
+		}
+
+		// The same as above, but for accepted channel request count. Note that we are interested
+		// only in confirmed open requests.
+		assert_eq!(
+			<Router as Store>::HrmpAcceptedChannelRequestCount::iter()
+				.map(|(k, _)| k)
+				.collect::<HashSet<_>>(),
+			<Router as Store>::HrmpOpenChannelRequests::iter()
+				.filter(|(_, v)| v.confirmed)
+				.map(|(k, _)| k.recipient)
+				.collect::<HashSet<_>>(),
+		);
+		for (channel_recipient, expected_num) in
+			<Router as Store>::HrmpAcceptedChannelRequestCount::iter()
+		{
+			let actual_num = <Router as Store>::HrmpOpenChannelRequests::iter()
+				.filter(|(ch, v)| ch.recipient == channel_recipient && v.confirmed)
+				.count() as u32;
+			assert_eq!(expected_num, actual_num);
+		}
+
+		assert_eq!(
+			<Router as Store>::HrmpCloseChannelRequests::iter()
+				.map(|(k, _)| k)
+				.collect::<HashSet<_>>(),
+			<Router as Store>::HrmpCloseChannelRequestsList::get()
+				.into_iter()
+				.collect::<HashSet<_>>(),
+		);
+
+		// A HRMP watermark can be None for an onboarded parachain. However, an offboarded parachain
+		// cannot have an HRMP watermark: it should've been cleanup.
+		assert_contains_only_onboarded(
+			<Router as Store>::HrmpWatermarks::iter().map(|(k, _)| k),
+			"HRMP watermarks should contain only onboarded paras",
+		);
+
+		// An entry in `HrmpChannels` indicates that the channel is open. Only open channels can
+		// have contents.
+		for (non_empty_channel, contents) in <Router as Store>::HrmpChannelContents::iter() {
+			assert!(<Router as Store>::HrmpChannels::contains_key(
+				&non_empty_channel
+			));
+
+			// pedantic check: there should be no empty vectors in storage, those should be modeled
+			// by a removed kv pair.
+			assert!(!contents.is_empty());
+		}
+
+		// Senders and recipients must be onboarded. Otherwise, all channels associated with them
+		// are removed.
+		assert_contains_only_onboarded(
+			<Router as Store>::HrmpChannels::iter().flat_map(|(k, _)| vec![k.sender, k.recipient]),
+			"senders and recipients in all channels should be onboarded",
+		);
+
+		// Check the docs for `HrmpIngressChannelsIndex` and `HrmpEgressChannelsIndex` in decl
+		// storage to get an index what are the channel mappings indexes.
+		//
+		// Here, from indexes.
+		//
+		// ingress         egress
+		//
+		// a -> [x, y]     x -> [a, b]
+		// b -> [x, z]     y -> [a]
+		//                 z -> [b]
+		//
+		// we derive a list of channels they represent.
+		//
+		//   (a, x)         (a, x)
+		//   (a, y)         (a, y)
+		//   (b, x)         (b, x)
+		//   (b, y)         (b, y)
```

Comment:
```suggestion
		// ingress         egress
		//
		// a -> [x, y]     x -> [a, b]
		// b -> [x, z]     y -> [a]
		//                 z -> [b]
		//
		// we derive a list of channels they represent.
		//
		//   (a, x)         (a, x)
		//   (a, y)         (a, y)
		//   (b, x)         (b, x)
		//   (b, z)         (b, z)
```

Looks like a typo in the conversion table: `b` maps to `z` in two cases, not `y`.